core/src/test/java/org/elasticsearch/VersionTests.java

```diff
@@ -352,4 +352,10 @@ public boolean isCompatible(Version left, Version right) {
         return result;
     }
 
+    private static final Version VERSION_5_1_0_UNRELEASED = Version.fromString("5.1.0");
+
+    public void testUnreleasedVersion() {
```

Comment:
I know it didn't have a comment before but I imagine it'd be nice to add a comment now. Something like "This exists because 5.1.0 was never released due to a mistake in the release process. This verifies that we never declare the version as "released" accidentally. It would never pass qa tests later on, but those come *very* far in the build and this is quick to check now."